File: Backend/src/controllers/authController.js
const authService = require('../services/authService');

const register = async (req, res, next) => {
    try {
        const { username, password } = req.body;
        if (!username || !password) {
            return res.status(400).json({ success: false, error: 'Username dan password wajib diisi.' });
        }
        
        const user = await authService.register(username, password);
        res.status(201).json({ success: true, data: user });
    } catch (error) {
        next(error);
    }
};

const login = async (req, res, next) => {
    try {
        const { username, password } = req.body;
        if (!username || !password) {
            return res.status(400).json({ success: false, error: 'Username dan password wajib diisi.' });
        }
        
        const result = await authService.login(username, password);
        res.status(200).json({ success: true, data: result });
    } catch (error) {
        next(error);
    }
};

module.exports = {
    register,
    login
};

File: Backend/src/main/java/com/rummikub/backend/controllers/AuthController.java
package com.rummikub.backend.controllers;

import com.rummikub.backend.models.User;
import com.rummikub.backend.services.AuthService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    @Autowired
    private AuthService authService;

    @PostMapping("/register")
    public ResponseEntity<?> register(@RequestBody Map<String, String> payload) {
        try {
            String username = payload.get("username");
            String password = payload.get("password");
            if (username == null || password == null) {
                return ResponseEntity.badRequest().body(Map.of("success", false, "error", "Username dan password wajib diisi."));
            }
            User user = authService.register(username, password);
            return ResponseEntity.status(201).body(Map.of("success", true, "data", user));
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(Map.of("success", false, "error", e.getMessage()));
        }
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody Map<String, String> payload) {
        try {
            String username = payload.get("username");
            String password = payload.get("password");
            if (username == null || password == null) {
                return ResponseEntity.badRequest().body(Map.of("success", false, "error", "Username dan password wajib diisi."));
            }
            Map<String, Object> result = authService.login(username, password);
            return ResponseEntity.ok(Map.of("success", true, "data", result));
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(Map.of("success", false, "error", e.getMessage()));
        }
    }
}
